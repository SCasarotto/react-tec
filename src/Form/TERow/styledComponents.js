import styled from 'styled-components'

export const Row = styled.div`
    ${(props) => {
        const { size, last } = props

        let width = '100%'
        switch (size) {
            case 'full':
                width = '100%'
                break
            case 'two-third':
                width = '66.66%'
                break
            case 'half':
                width = '50%'
                break
            case 'third':
                width = '33.33%'
                break
            case 'forth':
                width = '25%'
                break
            case 'condensed':
                width = 'auto'
                break
            default:
                break
        }

        return `
            position: relative;
            display: inline-block;
            vertical-align: top;
            margin-bottom: 10px;
            text-align: left;
            box-sizing: border-box;
            width: ${width};
            padding-left: ${size !== 'full' && last ? '5px' : '0px'};
            padding-right: ${size !== 'full' && !last ? '5px' : '0px'};
        `
    }}
`
