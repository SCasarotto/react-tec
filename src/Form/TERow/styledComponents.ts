import styled from 'styled-components'

export const Row = styled.div<{
	rowSize?: string
	last?: boolean
}>`
	position: relative;
	display: inline-block;
	vertical-align: top;
	margin-bottom: 10px;
	text-align: left;
	box-sizing: border-box;
	padding-right: ${(props) => (props.rowSize !== 'full' && !props.last ? '5px' : '0px')};

	${(props) => {
		const { rowSize } = props

		switch (rowSize) {
			case 'full':
				return 'width: 100%;'
			case 'two-third':
				return 'width: 66.66%'
			case 'half':
				return 'width: 50%'
			case 'third':
				return 'width: 33.33%;'
			case 'forth':
				return 'width: 25%;'
			case 'condensed':
				return 'width: auto;'
			default:
				return 'width: 100%;'
		}
	}}
`
