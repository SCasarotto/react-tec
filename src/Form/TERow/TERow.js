//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Row } from './styledComponents'

const TERow = (props) => <Row {...props} />

TERow.propTypes = {
	size: PropTypes.string,
	last: PropTypes.bool,
}

TERow.defaultProps = {
	size: 'full',
	last: false,
}

export default TERow
