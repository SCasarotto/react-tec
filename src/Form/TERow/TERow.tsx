import React from 'react'

import { Row } from './styledComponents'

export type TEFormRowSize = 'full' | 'two-third' | 'half' | 'third' | 'forth' | 'condensed'
export interface TERowCustomProps {
	rowSize?: TEFormRowSize
	last?: boolean
}
export interface TERowProps extends React.HTMLAttributes<HTMLDivElement>, TERowCustomProps {}
export const TERow = React.forwardRef<HTMLDivElement, TERowProps>((props, ref) => {
	const { rowSize = 'full', className = '', ...rest } = props
	return <Row className={`TERow ${className}`} rowSize={rowSize} ref={ref} {...rest} />
})
