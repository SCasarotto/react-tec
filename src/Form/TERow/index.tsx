import React from 'react'

import { Row } from './styledComponents'

export interface TERowCustomProps {
	rowSize?: string
	last?: boolean
}
export interface TERowProps
	extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
		TERowCustomProps {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
export const TERow: React.FC<TERowProps> = (props) => {
	const { rowSize = 'full', last = false, className = '', ...rest } = props
	return <Row className={`TERow ${className}`} rowSize={rowSize} last={last} {...rest} />
}
