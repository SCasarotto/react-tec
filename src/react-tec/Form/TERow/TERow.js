//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Row } from './styledComponents'

const TERow = (props) => {
    const { children, ...rest } = props

    return <Row {...rest}>{children}</Row>
}

TERow.propTypes = {
    size: PropTypes.string,
    last: PropTypes.bool,
    style: PropTypes.object,
    children: PropTypes.node,
}

TERow.defaultProps = {
    size: 'full',
    last: false,
}

export default TERow
