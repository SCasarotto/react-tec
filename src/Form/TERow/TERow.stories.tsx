import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TERow, TERowProps } from './TERow'

export default {
	title: 'Form/TERow',
	component: TERow,
	argTypes: {},
} as Meta

const Template: Story<TERowProps> = (args) => <TERow {...args} />

export const Default = Template.bind({})
Default.args = {}
