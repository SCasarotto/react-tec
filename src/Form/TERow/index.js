import TERow from './TERow'

export default TERow
