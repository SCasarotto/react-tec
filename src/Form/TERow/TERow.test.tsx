import React from 'react'
import { render } from '@testing-library/react'
import { TERow } from './TERow'

describe('TERow', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TERow />)
		expect(asFragment()).toMatchSnapshot()
	})
})
