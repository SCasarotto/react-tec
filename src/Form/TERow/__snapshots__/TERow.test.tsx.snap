// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TERow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

<div
    class="c0 TERow "
  />
</DocumentFragment>
`;
